sway-core/src/declaration_engine/declaration_id.rs

```diff
@@ -1,7 +1,7 @@
 use std::fmt;
 
 /// An ID used to refer to an item in the [DeclarationEngine]
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
```

Comment:
Can you remove `Hash, PartialEq, Eq`? Those will need to be special cased I think.

Reply:
@tritao you should be able to pull in  #2675 and solve this now :smile: